### PR TITLE
RES-1483: Make sure checkpoint used to restore does not override model initial checkpoint state

### PR DESCRIPTION
**Pull request by lscheinkman**

### CODE REVIEW

File: nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py [hunk 1]
```diff
@@ -83,24 +83,27 @@ def _setup(self, config):
...
-        if self._checkpoint_file is not None:
-            with open(self._checkpoint_file, mode="rb") as f:
+        self._restored = False
+        if self.restore_checkpoint_file is not None:
+            with open(self.restore_checkpoint_file, mode="rb") as f:
                 state = pickle.load(f)
                 self._restore(state)
                 self._restored = True
 
+        elif config.get("checkpoint_at_init", False):
```
Comment (subutai) [inline nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py:103]: For consistency, wouldn't we still want to save the checkpoint at init if that flag is set to True?

Comment (lscheinkman) [reply nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py:103]: Are you sure? It would just be a copy of the original checkpoint?

**Comment by lscheinkman** [comment]
> How would we run the _continue experiments? Would we specify the checkpoint on the command line?

`python run.py -c checkpoint` to resume from a specific checkpoint file or `python run.py --restore` to continue training from the last known checkpoint.

File: nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py (hunk 1 above)
Comment (subutai) [reply nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py:103]: It would, just wondering for consistency. I'm fine either way.

**Pull request merged**